File: dart/lib/model/v3_siri_downstream_subscription.dart
part of swagger.api;

class V3SiriDownstreamSubscription {
  
  String subscriberRef = null;

  String subscriptionRef = null;

  int messageType = null;
  //enum messageTypeEnum {  0,  1,  };

  int siriFormat = null;
  //enum siriFormatEnum {  0,  1,  };

  String siriVersion = null;

  String consumerAddress = null;

  DateTime initialTerminationTime = null;

  DateTime validityPeriodStart = null;

  DateTime validityPeriodEnd = null;

  String previewInterval = null;

  List<V3SiriDownstreamSubscriptionTopic> topics = [];

  V3SiriDownstreamSubscription();

  @override
  String toString() {
    return 'V3SiriDownstreamSubscription[subscriberRef=$subscriberRef, subscriptionRef=$subscriptionRef, messageType=$messageType, siriFormat=$siriFormat, siriVersion=$siriVersion, consumerAddress=$consumerAddress, initialTerminationTime=$initialTerminationTime, validityPeriodStart=$validityPeriodStart, validityPeriodEnd=$validityPeriodEnd, previewInterval=$previewInterval, topics=$topics, ]';
  }

  V3SiriDownstreamSubscription.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    subscriberRef = json['subscriber_ref'];
    subscriptionRef = json['subscription_ref'];
    messageType = json['message_type'];
    siriFormat = json['siri_format'];
    siriVersion = json['siri_version'];
    consumerAddress = json['consumer_address'];
    initialTerminationTime = json['initial_termination_time'] == null ? null : DateTime.parse(json['initial_termination_time']);
    validityPeriodStart = json['validity_period_start'] == null ? null : DateTime.parse(json['validity_period_start']);
    validityPeriodEnd = json['validity_period_end'] == null ? null : DateTime.parse(json['validity_period_end']);
    previewInterval = json['preview_interval'];
    topics = V3SiriDownstreamSubscriptionTopic.listFromJson(json['topics']);
  }

  Map<String, dynamic> toJson() {
    return {
      'subscriber_ref': subscriberRef,
      'subscription_ref': subscriptionRef,
      'message_type': messageType,
      'siri_format': siriFormat,
      'siri_version': siriVersion,
      'consumer_address': consumerAddress,
      'initial_termination_time': initialTerminationTime == null ? '' : initialTerminationTime.toUtc().toIso8601String(),
      'validity_period_start': validityPeriodStart == null ? '' : validityPeriodStart.toUtc().toIso8601String(),
      'validity_period_end': validityPeriodEnd == null ? '' : validityPeriodEnd.toUtc().toIso8601String(),
      'preview_interval': previewInterval,
      'topics': topics
     };
  }

  static List<V3SiriDownstreamSubscription> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriDownstreamSubscription>() : json.map((value) => new V3SiriDownstreamSubscription.fromJson(value)).toList();
  }

  static Map<String, V3SiriDownstreamSubscription> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriDownstreamSubscription>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriDownstreamSubscription.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriDownstreamSubscription.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriDownstreamSubscription {
    /// <summary>
    /// Gets or Sets SubscriberRef
    /// </summary>
    [DataMember(Name="subscriber_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "subscriber_ref")]
    public string SubscriberRef { get; set; }

    /// <summary>
    /// Gets or Sets SubscriptionRef
    /// </summary>
    [DataMember(Name="subscription_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "subscription_ref")]
    public string SubscriptionRef { get; set; }

    /// <summary>
    /// Gets or Sets MessageType
    /// </summary>
    [DataMember(Name="message_type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "message_type")]
    public int? MessageType { get; set; }

    /// <summary>
    /// Gets or Sets SiriFormat
    /// </summary>
    [DataMember(Name="siri_format", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "siri_format")]
    public int? SiriFormat { get; set; }

    /// <summary>
    /// Gets or Sets SiriVersion
    /// </summary>
    [DataMember(Name="siri_version", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "siri_version")]
    public string SiriVersion { get; set; }

    /// <summary>
    /// Gets or Sets ConsumerAddress
    /// </summary>
    [DataMember(Name="consumer_address", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "consumer_address")]
    public string ConsumerAddress { get; set; }

    /// <summary>
    /// Gets or Sets InitialTerminationTime
    /// </summary>
    [DataMember(Name="initial_termination_time", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "initial_termination_time")]
    public DateTime? InitialTerminationTime { get; set; }

    /// <summary>
    /// Gets or Sets ValidityPeriodStart
    /// </summary>
    [DataMember(Name="validity_period_start", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "validity_period_start")]
    public DateTime? ValidityPeriodStart { get; set; }

    /// <summary>
    /// Gets or Sets ValidityPeriodEnd
    /// </summary>
    [DataMember(Name="validity_period_end", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "validity_period_end")]
    public DateTime? ValidityPeriodEnd { get; set; }

    /// <summary>
    /// Gets or Sets PreviewInterval
    /// </summary>
    [DataMember(Name="preview_interval", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "preview_interval")]
    public string PreviewInterval { get; set; }

    /// <summary>
    /// Gets or Sets Topics
    /// </summary>
    [DataMember(Name="topics", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "topics")]
    public List<V3SiriDownstreamSubscriptionTopic> Topics { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriDownstreamSubscription {\n");
      sb.Append("  SubscriberRef: ").Append(SubscriberRef).Append("\n");
      sb.Append("  SubscriptionRef: ").Append(SubscriptionRef).Append("\n");
      sb.Append("  MessageType: ").Append(MessageType).Append("\n");
      sb.Append("  SiriFormat: ").Append(SiriFormat).Append("\n");
      sb.Append("  SiriVersion: ").Append(SiriVersion).Append("\n");
      sb.Append("  ConsumerAddress: ").Append(ConsumerAddress).Append("\n");
      sb.Append("  InitialTerminationTime: ").Append(InitialTerminationTime).Append("\n");
      sb.Append("  ValidityPeriodStart: ").Append(ValidityPeriodStart).Append("\n");
      sb.Append("  ValidityPeriodEnd: ").Append(ValidityPeriodEnd).Append("\n");
      sb.Append("  PreviewInterval: ").Append(PreviewInterval).Append("\n");
      sb.Append("  Topics: ").Append(Topics).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
